manifests/configuration.pp

```diff
@@ -0,0 +1,110 @@
+# Manage sysctl variable values.
+#
+# Parameters:
+#  $value:
+#    The value for the sysctl parameter. Mandatory, unless $ensure is 'absent'.
+#  $prefix:
+#    Optional prefix for the sysctl.d file to be created. Default: none.
+#  $ensure:
+#    Whether the variable's value should be 'present' or 'absent'.
+#    Defaults to 'present'.
+#
+# Sample Usage :
+#  sysctl::configuration { 'net.ipv6.bindv6only': value => '1' }
+define sysctl::configuration (
+  String $variable               = $title,
+  String $ensure                 = 'present',
+  Variant[String,Undef] $value   = undef,
+  Variant[String,Undef] $prefix  = undef,
+  String $suffix                 = '.conf',
+  Variant[String,Undef] $comment = undef,
+  Variant[String,Undef] $content = undef,
+  Variant[String,Undef] $source  = undef,
+  Boolean $enforce               = true,
+  String $sysctl_binary          = '/sbin/sysctl',
+  String $sysctl_dir_path        = '/etc/sysctl.d',
+) {
+
+  # If we have a prefix, then add the dash to it
+  if $prefix {
+    $_sysctl_d_file = "${prefix}-${variable}${suffix}"
+  } else {
+    $_sysctl_d_file = "${variable}${suffix}"
```

Comment:
hm the new review system is nice but i don't get notifiaction emails..

i just installed your fork, which makes again a wrong config. 
```
'ipv6_disable_net.ipv6.conf.eth0.disable_ipv6': {  variable: 'net.ipv6.conf.eth0.disable_ipv6', value: '1', prefix: '90', comment: 'debian disable ipv6 https://wiki.debian.org/DebianIPv6' }
```
creates 
```
90-net.ipv6.conf.eth0.disable_ipv6.conf
```
with my fork it creates the correct file, names:
```
90-ipv6_disable_net.ipv6.conf.eth0.disable_ipv6.conf
```
also tried your example with my fork:
```
#filename: disable-ipv4-forwarding.conf
net.ipv4.ip_forward = 0
```
it makes no sense to me, if there is a puppet variable that is named `$_sysctl_d_file` (file!), it should handle the name of the file, which is set via the `name`/`title` parameter and should not be named after the variable.
can you try my fork please? what os & what puppet version are you using?